lib/hooks/useGCFormContext.tsx

```diff
@@ -56,6 +65,7 @@ export const GCFormsProvider = ({
 
       if (typeof nextAction === "string") {
         setCurrentGroup(nextAction);
+        addHistory(nextAction);
```

Comment:
Instead of coupling here, this could be moved to the Next button but then some of this logic would probably need to be duplicated.